test/cpp/jit/test_gpu_shift.cpp

```diff
@@ -1670,6 +1670,101 @@ TEST(NVFuserTest, FusionShift5ptStencilChain_CUDA) {
   testValidate(&fusion, outputs, inputs, {ref}, __LINE__, __FILE__);
 }
 
+// 3x3 max pooling
+TEST(NVFuserTest, FusionMaxPooling_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  // Format: CHW
+  auto inp = makeSymbolicTensor(3);
+  fusion.addInput(inp);
+
+  // 3x3 pooling of the HW spatial domain
+  std::vector<std::vector<int>> offsets;
+  for (int i = -1; i <= 1; ++i) {
+    for (int j = -1; j <= 1; ++j) {
+      if (i == 0 && j == 0) {
+        continue;
+      }
+      offsets.push_back({i, j});
+    }
+  }
+
+  std::vector<TensorView*> inp_tile({inp});
+  for (auto offset : offsets) {
+    offset.insert(offset.begin(), 0);
+    inp_tile.push_back(shift(inp, offset));
+  }
+
+  TensorView* max_tensor = nullptr;
+  ;
+  for (auto tv : inp_tile) {
+    if (max_tensor == nullptr) {
+      max_tensor = tv;
+    } else {
+      max_tensor = binaryOp(BinaryOpType::Max, max_tensor, tv);
+    }
+  }
+
+  fusion.addOutput(max_tensor);
+
+  ////////////////////////////////////
+
+  // Cache the input and weight tensors
+  auto inp_cache = inp->cache_after();
+
+  // Tiling the spatial domain
+  const int tile_x = 32;
+  const int tile_y = 8;
+
+  max_tensor->split(-2, tile_y);
+  max_tensor->axis(-2)->parallelize(ParallelType::TIDy);
+  max_tensor->split(-1, tile_x);
+  max_tensor->axis(-1)->parallelize(ParallelType::TIDx);
+  max_tensor->reorder({{-3, -2}});
+
+  inp_cache->computeAt(max_tensor, 3);
```

Comment:
vShould we work on shift support for compute at most inlined and best effort?

Reply:
Eventually, yes definitely. At this point, I think it's more important to focus more on more primitive interfaces since they don't work, none of the more powerful interfaces like best effort should work.